### PR TITLE
feat: add query while reloading example

### PR DESCRIPTION
**Pull request by cristianmtr**
Add example to show querying while reloading on the Wikipedia example

Part of https://github.com/jina-ai/internal-tasks/issues/40

Minor fixes for wikipedia example

### CODE REVIEW

File: wikipedia-sentences-query-while-indexing/app.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+SHARDS = 3  # change this if you change pods/query_indexer.yml
+DUMP_RELOAD_INTERVAL = 20  # time between dump - rolling update calls
+DATA_FILE = 'data/toy.txt'  # change this if you get the full data
+DOCS_PER_ROUND = 5  # nr of documents to index in each round
+DBMS_REST_PORT = '9000'  # REST port of DBMS Flow, defined in flows/dbms.yml
+QUERY_REST_PORT = '9001'  # REST port of Query Flow, defined in flows/query.yml
+
+
+def query_restful():
+    while True:
```
Comment (JoanFM) [inline wikipedia-sentences-query-while-indexing/app.py:34]: query restful shpuld not wait for the user to send queries? how will this work from docker? can docker capture user's std input?

Comment (cristianmtr) [reply wikipedia-sentences-query-while-indexing/app.py:34]: Talked to Fionn, won't be building a docker for this example

**Comment by nan-wang** [comment]
@cristianmtr Can we add a test for this? Please check out the tests at https://github.com/jina-ai/examples/blob/master/wikipedia-sentences/tests/test_wikipediasearch.py

File: wikipedia-sentences-query-while-indexing/app.py (hunk 1 above)
Comment (JoanFM) [reply wikipedia-sentences-query-while-indexing/app.py:34]: then remove the Dockerfile, but anyway, if u do query restful I would expect people to send rest queries and not typing right?

Comment (cristianmtr) [reply wikipedia-sentences-query-while-indexing/app.py:34]: yes, it's removed and the README includes instructions on how to do REST queries with curl etc.

**Comment by cristianmtr** [comment]
> @cristianmtr Can we add a test for this? Please check out the tests at [https://github.com/jina-ai/examples/blob/master/wikipedia-sentences/tests/test_wikipediasearch.py](https://github.com/jina-ai/examples/blob/master/wikipedia-sentences/tests/test_wikipediasearch.py?rgh-link-date=2021-05-12T08%3A58%3A45Z)

Yes, working on it. 

It's not as easy as the other examples, as it requires `jinad` and `python app.py -t flows` to run while the query client is being tested. But will try to figure it out

**Review by JoanFM — APPROVED** [review]

File: wikipedia-sentences-query-while-indexing/README.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
+# Run the Querying While Indexing Wikipedia Search Example
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:1]: Suggestion: Shorten to just _Querying While Indexing in the Wikipedia Search Example_

File: wikipedia-sentences-query-while-indexing/README.md [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+You can check the `flows` and `pods` directories for the changes to the files.
+_____
+
+## 🐍 Build the app with Python
+
+These instructions explain how to run the example yourself and deploy it with Python. 
+
+### 🗝️ Requirements
+
+1. Have a working Python 3.7 environment.
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:43]: I think we can also use 3.8 now right?

File: wikipedia-sentences-query-while-indexing/README.md [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+
+```sh
+pip install -r requirements.txt
+```
+
+In order to run the example you will need to do the following:
+
+### 📥 Step 2. Download your data to search (Optional)
+
+The example repo includes a small subset of the Wikipedia dataset, for quick testing. You can just use that.
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:73]: Suggestion: This repo includes

File: wikipedia-sentences-query-while-indexing/README.md [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+### 📥 Step 2. Download your data to search (Optional)
+
+The example repo includes a small subset of the Wikipedia dataset, for quick testing. You can just use that. 
+
+If you want to use the entire dataset, run `bash get_data.sh` and then modify the `DATA_FILE` constant (in `app.py`) to point to that file.
+
+### 🏃 Step 3. Running the Flows
+
+1. In one terminal session, run `jinad`.
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:79]: suggestion: run the command `jinad`

File: wikipedia-sentences-query-while-indexing/README.md [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+    `JinaD` is our platform for running Jina services (Flows, Pods) remotely, wherever you want to run them. In this example, we use `JinaD` to serve the two Flows (Index and Query). You can read more about it [in the docs](https://docs.jina.ai/chapters/remote/jinad).
+
+2. In another terminal, run `python app.py -t flows`
+
+    This will create the two Flows, and then repeatedly do the following, every 20 seconds:
+
+    1. index 5 Documents
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:87]: Capital letters for Index, send, and send.

File: wikipedia-sentences-query-while-indexing/README.md [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+    2. send a `DUMP` request to the DBMS (Index) Flow to dump its data to a specific location
+    3. send a `ROLLING_UPDATE` request to the Query Flow to take down its Indexers and start them again, with the new data located at the respective path
+    
+    **Notice** the logs of the operations.
+
+    **Warning**: the data file is limited to 200 documents. Once that is exhausted, the process will terminate. If you want to use the entire dataset, run `bash get_data.sh` and then modify the `DATA_FILE` constant to point to that file.
+
+### 🔎 Step 4: Query your data
+
+Finally, in another terminal, run `python app.py -t client`
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:97]: in a third terminal 

- This clarify to the user how many terminals they should have open.

File: wikipedia-sentences-query-while-indexing/README.md [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+Optionally, if you have [`jq`](https://stedolan.github.io/jq/), you can just get the text of the matches of the query:
+
+```sh
+curl -X POST -d '{"data": [{"text":"hello world"}]}' http://0.0.0.0:9001/search | jq -r '.search.docs[] | .matches[] | .text'
+```
+
+## 🔮 Overview of the files
+
+*Add a list with all folders/files in the example:*
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:118]: Remove

File: wikipedia-sentences-query-while-indexing/README.md [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+| --- 📃 `query.yml`     | YAML file to configure Querying Flow                                                                             |
+| 📂 `pods/`           | Folder to store Pod configuration                                                                                |
+| --- 📃 `dbms_indexer.yml`   | YAML file to configure the DBMS Indexer                                                                               |
+| --- 📃 `encoder.yml`   | YAML file to configure encoder Pod                                                                               |
+| --- 📃 `query_indexer.yml`   | YAML file to configure the Query Indexer                                                                               |
+| 🐍 `app.py`      | Code file for the example   |
+
+
+_________
+
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:134]: Missing: 
Flow Digrams for dbms and query flows

File: wikipedia-sentences-query-while-indexing/README.md [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+If you have any issues following this guide, you can always get support from our [Slack community](https://join.slack.com/t/jina-ai/shared_invite/zt-dkl7x8p0-rVCv~3Fdc3~Dpwx7T7XG8w) .
+
+## 👩‍👩‍👧‍👦 Community
+
+- [Slack channel](slack.jina.ai) - a communication platform for developers to discuss Jina.
+- [LinkedIn](https://www.linkedin.com/company/jinaai/) - get to know Jina AI as a company and find job opportunities.
+- [![Twitter Follow](https://img.shields.io/twitter/follow/JinaAI_?label=Follow%20%40JinaAI_&style=social)](https://twitter.com/JinaAI_) - follow us and interact with us using hashtag `#JinaSearch`.  
+- [Company](https://jina.ai) - know more about our company. We are fully committed to open-source!
+
+## 🦄 License
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:148]: Should we add a troubleshooting section containing the two issues I found? 
1. When running JinaD from a virtual environment. Make sure the directory path in your terminal matches the JinaD directory path printed.  
2. On some Mac machines, you may have to run JinaD with sudo command.

File: wikipedia-sentences-query-while-indexing/README.md [hunk 11]
```diff
@@ -0,0 +1,152 @@
+# Run the Querying While Indexing Wikipedia Search Example
+
```
Comment (FionnD) [inline wikipedia-sentences-query-while-indexing/README.md:2]: Missing: Table of contents

**Review by FionnD — CHANGES_REQUESTED** [review]
Thanks for the late-night Friday tech support

**Review by FionnD — DISMISSED** [review]
LGTM! Please update the README of the repo to include your example. https://github.com/jina-ai/examples#--advanced-examples

**Review by FionnD — DISMISSED** [review]

**Review by FionnD — APPROVED** [review]

**Review by FionnD — APPROVED** [review]

**Pull request merged**